js/songs/edit.controller.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+
+(function(){
+  angular
+  .module("songs")
+  .controller("SongEditController", [
+    "SongFactory",
+    "$stateParams",
+    SongEditControllerFunction
+  ]);
+
+  function SongEditControllerFunction(SongFactory, $stateParams) {
+    this.song = SongFactory.get({id: $stateParams.id});
+    this.update = function() {
+      this.song.$update({id: $stateParams.id});
```

Comment:
after we make an `update` request. We can call `$state.go` and pass in the name of a state to go to.  If you want to trigger a page "refresh" you can pass that method call some additional options, such as any important parameters or options on how to reload